File: engine/model/src/main/scala/tools/packet/guild/GuildEmblemChange.scala
package tools.packet.guild

import net.opcodes.SendOpcode
import tools.packet.PacketInput

class GuildEmblemChange(private var _guildId: Int, private var _background: Short, private var _backgroundColor: Byte,
                        private var _logo: Short, private var _logoColor: Byte) extends PacketInput {
  def guildId: Int = _guildId

  def background: Short = _background

  def backgroundColor: Byte = _backgroundColor

  def logo: Short = _logo

  def logoColor: Byte = _logoColor

  override def opcode(): SendOpcode = SendOpcode.GUILD_OPERATION
}

File: engine/engine-model/src/main/java/tools/packet/guild/GuildEmblemChange.java
package tools.packet.guild;

import net.opcodes.SendOpcode;
import tools.packet.PacketInput;

public record GuildEmblemChange(Integer guildId, Short background, Byte backgroundColor, Short logo,
                                Byte logoColor) implements PacketInput {
   @Override
   public SendOpcode opcode() {
      return SendOpcode.GUILD_OPERATION;
   }
}